src/chess/board.cc

```diff
@@ -982,8 +982,36 @@ void ChessBoard::SetFromFen(const std::string& fen, int* rule50_ply,
   string en_passant;
   int rule50_halfmoves;
   int total_moves;
-  fen_str >> board >> who_to_move >> castlings >> en_passant >>
-      rule50_halfmoves >> total_moves;
+  fen_str >> board;
+  if (fen_str.eof()) {
```

Comment:
Does it really work?.. eof bit is set when there is an attempt to read behind end of string, not when it's just reached.
So if fen = `"k/1R//K ";` (note trailing space, I thought it wouldn't work)

Reply:
The unit test would fail when directly passing in a string with trailing space to `SetFromFen` but `ParseCommand` tokenizes and combines with a single space https://github.com/LeelaChessZero/lc0/blob/9b3c7992aef13a477b084fdaf3304fea5ed9e4b4/src/chess/uciloop.cc#L81-L87

Reply:
What happens with a >> std::ws added after each section read?

Reply:
ahh okay that shouldn't work would need to be if (fen_str.eof() || fen_str >> std::ws && fen_str.eof()) or something.

Reply:
Added a commit that removes trailing whitespace so plain `if (fen_str.eof())` will work and spaces ahead / between fields will be handled normally with `skipws`